src/pages/DatasetStatistics.jsx

```diff
@@ -9,15 +9,25 @@ import { formatDate } from '../libs/utils';
 
 const LINE = <div style={{ borderTop: '1px solid #BABEC1', height: 0 }} />;
 
+const extractIdentifier = (params) => {
+  let datasetIdentifier = params.datasetIdentifier;
+  if (params.duosId) {
+    datasetIdentifier = "DUOS-" + params.duosId;
+  }
+  return datasetIdentifier.toUpperCase();
+}
+
 export default function DatasetStatistics(props) {
-  const datasetId = props.match.params.datasetId;
+  const datasetIdentifier = extractIdentifier(props.match.params);
   const [dataset, setDataset] = useState();
   const [dars, setDars] = useState();
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    setData(datasetId);
-  }, [datasetId]);
+    DataSet.getDatasetByDatasetIdentifier(datasetIdentifier).then((dataset) => {
+      setData(dataset.dataSetId);
```

Comment:
Fails silently for incorrect dataset identifiers. Should we handle errors here as well?